### ISSUE TITLE
failed to delete securityGroup: timed out waiting for the condition

### ISSUE THREAD

**Issue by irl-segfault** [kubernetes-sigs/aws-load-balancer-controller#3015]
**Describe the bug**
Deleting an ingress fails to delete some of the constituent components.  Specifically, the controller can not delete the security group for the ALB.  I believe this is because the controller did not adequately delete ingress rules in the target security group for allowing incoming traffic from the ELB security group (which is the one attempting to be deleted.)

```
failed to delete securityGroup: timed out waiting for the condition
```

**Steps to reproduce**
not exactly sure how to repro -- the controller got into a bad state somehow.  Wish I could provide concrete repro steps, but it involved overwriting a Helm chart with a new one.  

**Expected outcome**
The ingress gets deleted correctly. 

**Environment**
EKS v1.20

* AWS Load Balancer controller version
* v1.2.4
* Kubernetes version
* 1.20
* Using EKS (yes/no), if so version?
Yes, v1.20

**Additional Context**:

I've checked the permissions on the aws-load-balancer-controller, and they are all good.  To figure out what's going on, I tried to delete the security group in question in the console. It can not be deleted because the ingress rule allowing traffic from the ELB sg into the worker node SG is still present on the worker node SG.  It was not deleted by the controller, and therefore the ELB sg can not be deleted.  I believe this is the cause.  The error message about "timed out waiting for condition" made me think it was the issue around the ingress rule not being tagged properly with the `elbv2.k8s.aws/targetGroupBinding=shared` tag mentioned in the [migration guide](https://github.com/kubernetes-sigs/aws-load-balancer-controller/blob/v2.4.2/docs/deploy/upgrade/migrate_v1_v2.md#backwards-compatibility) but that's not it either.  Quite stumped!

I've attached a screenshot of some of the logs we're seeing coming out of the controller.  Notably, I don't see logs for the deletion of the ingress rule, which I believe to be the cause.  Is this a known issue?
<img width="546" alt="Screen Shot 2023-01-31 at 4 18 29 PM" src="https://user-images.githubusercontent.com/47435467/215914757-579a18cb-5e7c-41ba-8752-8d8bf53cc8cb.png">

**Comment by oliviassss** [comment]
Based on the screenshot, I didn't see the listener rules are getting deleted, however, they did get modified. Do you have other ingresses that are using the listener rules?

**Comment by irl-segfault** [comment]
are "listener rules" the same thing as "security group inbound rules?" 

I eneded up just deleting the inbound rule allowing traffic from ELB sg -> EKS sg.  Doing that eliminated the deletion constraint on the ELB security group, and the controller successfully tore down what remained.  The open question is why that rule didn't get deleted on its own -- I'm assuming this is some edge case here around deleting / recreating ingress objects in quick succession in the same namespace with similar specs.  Not sure.  Closing this out but hopefully someone with the same issue can find this and learn that the solution is to manually delete the inbound sg rule on the EKS cluster sg.  Thanks!
